conbench/job.py

```diff
@@ -58,15 +59,20 @@ class CacheDict(TypedDict):
 
 # Fetching one million items from a sample DB takes ~1 minute on my machine
 # (the `results = Session.scalars(....all())` call takes that long.
-def _fetch_and_cache_most_recent_results(n=0.07 * 10**6) -> None:
+def _fetch_and_cache_most_recent_results(n=0.06 * 10**6) -> None:
     log.debug(
         "BMRT cache: keys in cache: %s",
         len(_cache_bmrs["by_id"]),
     )
     t0 = time.monotonic()
 
+    # Also fetch hardware from associated Run, so that result.run is in cache,
+    # too, and so that result.run.hardware is a quick lookup.
     results = Session.scalars(
-        select(BenchmarkResult).order_by(BenchmarkResult.timestamp.desc()).limit(n)
+        select(BenchmarkResult)
```

Comment:
Maybe a good TODO: The query will take less time and memory if you only select the fields you'll need later; i.e. something like this. It will also be guaranteed to not hit the DB again later if you access attributes of the result rows.
```py
select(
  BenchmarkResult.id.label("result_id"),
  Hardware.hash.label("hardware_hash"),
  ...,
  Case.name.label("benchmark_name"),
  Case.tags.label("case_tags"),
  Case.id.label("case_id"),
)
```
Ah, but then you'd lose the ability to use the new `@property` methods on the BenchmarkResult object. Maybe it's worth it?

Reply:
Right! I also noted some immediate tuning ideas down in https://github.com/conbench/conbench/pull/1085 in the description. The good thing is that we _know_ that a lot can be done to make this quite a bit better/smarter in terms of resource usage.